server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/factory/DeployedCacheStoreFactory.java

```diff
@@ -18,14 +17,22 @@
  */
 public class DeployedCacheStoreFactory implements CacheStoreFactory {
 
-   private Map<String, DeployedCacheStoreMetadata> deployedCacheStores = Collections.synchronizedMap(new HashMap<String, DeployedCacheStoreMetadata>());
+   private static final int TIMEOUT = 20;
+   private static final TimeUnit TIMEOUT_UNITS = TimeUnit.SECONDS;
+
+   private PromiseMap<String, DeployedCacheStoreMetadata> deployedCacheStores = new PromiseMap<>();
```

Comment:
I personally think just using a normal ConcurrentMap<K, CompletableFuture<V>> directly would be much more clear  in how this would work.  Then all the timeouts are handled in this method as well.